gamersnet_backend/persistence/posts.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+let MongoDB = require('./mongodb');
+
+async function getAllPosts() {
+  // connect wait for server to connect to db
+  let db = await MongoDB.open();
+
+  // once it connected, get the "posts" collection (aka a table in SQL)
+  let posts = db.collection('posts');
+
+  // wait for the server to find all posts and return as an array
+  let result = await posts.find({});
+  return result.toArray();
+}
+
+//assuming posts expire if the gametime is older than current time
+//so get posts for games scheduled today or in future
+async function getValidPosts() {
+    let db = await MongoDB.open();
+    let posts = db.collection('posts');
+    
+    let result = await posts.find({ gameTimeUTC: {$gte: new Date()}});
+
+    return result.toArray();
+}
+
+
+/**
+ * Adds a post to the db
+ * All parameters should be string type to be consistent and avoid confusion
+ * @param {*} username name of the owner of this post
+ * @param {*} description 
+ * @param {*} gameName 
+ * @param {*} numPlayers 
+ * @param {*} gameTimeUTC when it will be played
+ * @param {*} duration how long will it be played
+ * @param {*} location location of game
+ */
+async function addPost(username, description, gameName, numPlayers, gameTimeUTC, duration, location) {
+  // wait for db connection and get users collection
+  let db = await MongoDB.open();
+
+  let posts = db.collection('posts');
+
+  return await posts.insertOne({
+    username: username,// intended to link to existing users in db
```

Comment:
Consider linking posts to users by the user's MongoDB document `$_id` instead of username

Reply:
I did go by the ObjectID before, but removed it later. I'll add that back in.

Reply:
just to make things easier on the frontend, do I keep the username as it is? since posts will display usernames too. So it is part of the post content.

Reply:
I was thinking to remove the username and just keep the ID since that links back to the user document. This way if the username changes for whatever reason, there will still be a link back to who made it. Then when displaying the username on the frontend, the route can just get the user document based on ID, which already exists as a function in /users/ I think